### PR TITLE
Fill in information about units of Intel performance counters.

### PR DESCRIPTION
**Pull request by marcinslusarz**
Please don't merge until https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/8580 lands.

### CODE REVIEW

File: renderdoc/driver/ihv/intel/intel_gl_counters.cpp [hunk 1]
```diff
@@ -104,6 +104,25 @@ void IntelGlCounters::addCounter(const IntelGlQuery &query, GLuint counterId)
...
+  if(counter.desc.description.contains("Unit: cycles."))
+    counter.desc.unit = CounterUnit::Cycles;
+  else if(counter.desc.description.contains("Unit: bytes."))
+    counter.desc.unit = CounterUnit::Bytes;
+  else if(counter.desc.description.contains("Unit: percent."))
+    counter.desc.unit = CounterUnit::Percentage;
+  else if(counter.desc.description.contains("Unit: Hz."))
+    counter.desc.unit = CounterUnit::Hertz;
+  else if(counter.desc.description.contains("Unit: ns."))
+  {
```
Comment (baldurk) [inline renderdoc/driver/ihv/intel/intel_gl_counters.cpp:147]: Style nit: if you need braces for any branch of an if/else chain please add it for all of them.

File: renderdoc/driver/ihv/intel/intel_gl_counters.cpp [hunk 2]
```diff
@@ -104,6 +104,25 @@ void IntelGlCounters::addCounter(const IntelGlQuery &query, GLuint counterId)
   counter.desc.resultType = glToRdcCounterType(counter.dataType);
   counter.desc.unit = CounterUnit::Absolute;
 
+  if(counter.desc.description.contains("Unit: cycles."))
```
Comment (baldurk) [inline renderdoc/driver/ihv/intel/intel_gl_counters.cpp:130]: I'm not a big fan of this kind of string compare especially when it's not spec'd and could change. At least this falls back to unitless, but is there any alternative to actually query the type as an enum from the driver?

**Review by baldurk — CHANGES_REQUESTED** [review]
I've left a couple of minor notes, but overall this is probably fine - though as I mention I'd prefer a way to obtain the units that doesn't need string compares.

In future though please don't open PRs that you aren't ready to merge. I'll leave this open this time, though I'd ask that you comment here when you're ready for it to be merged.

File: renderdoc/driver/ihv/intel/intel_gl_counters.cpp (hunk 2 above)
Comment (marcinslusarz) [reply renderdoc/driver/ihv/intel/intel_gl_counters.cpp:130]: Yeah, I'm not happy about this either, but doing it properly would require adding a new extension with a new API and I don't think it's worth it for such a minor change. I think we'll need more things to fix/add for a new extension.

Comment (baldurk) [reply renderdoc/driver/ihv/intel/intel_gl_counters.cpp:130]: Fair enough yeh, I thought that might be the case.

**Comment by marcinslusarz** [comment]
This PR is ready to be merged.

**Pull request merged**